pkg/crane/pull.go

```diff
@@ -38,19 +38,36 @@ func NewCmdPull() *cobra.Command {
 
 func pull(_ *cobra.Command, args []string) {
 	src, dst := args[0], args[1]
-	// TODO: Why is only tag allowed?
-	t, err := name.NewTag(src, name.WeakValidation)
+
+	ref, err := name.ParseReference(src, name.WeakValidation)
 	if err != nil {
 		log.Fatalf("parsing tag %q: %v", src, err)
 	}
-	log.Printf("Pulling %v", t)
+	log.Printf("Pulling %v", ref)
 
-	i, err := remote.Image(t, remote.WithAuthFromKeychain(authn.DefaultKeychain))
+	i, err := remote.Image(ref, remote.WithAuthFromKeychain(authn.DefaultKeychain))
 	if err != nil {
-		log.Fatalf("reading image %q: %v", t, err)
+		log.Fatalf("reading image %q: %v", ref, err)
+	}
+
+	// WriteToFile wants a tag to write to the tarball, but we might have
+	// been given a digest.
+	// If the original ref was a tag, use that. Otherwise, if it was a
+	// digest, remove the digest from the ref instead.
```

Comment:
FWIW in the python lib this became "i-was-a-digest": https://github.com/google/containerregistry/blob/63ff239fa771e81237804b24ae5ff237cedbf42a/tools/docker_puller_.py#L58-L74

Removing the digest just makes this `latest` in my head... not sure what the actual behavior is, but a sentinel to denote that it was pulled by digest seems better.

Reply:
That's a really good idea. Done.